ddtrace/tracer/option.go

```diff
@@ -322,15 +330,6 @@ func newConfig(opts ...StartOption) *config {
 	return c
 }
 
-// defaultHTTPClient returns the default http.Client to start the tracer with.
-func defaultHTTPClient() *http.Client {
-	if _, err := os.Stat(defaultSocketAPM); err == nil {
-		// we have the UDS socket file, use it
-		return udsClient(defaultSocketAPM)
```

Comment:
⚠️ Why is the existing `udsClient` function not removed? `WithUDS` is still using it, but I don't see why this wasn't switched to the new `UDSClient` method?